src/v/cloud_storage/topic_manifest.cc

```diff
@@ -235,12 +235,15 @@ topic_manifest::topic_manifest()
   : _topic_config(std::nullopt) {}
 
 void topic_manifest::do_update(const topic_manifest_handler& handler) {
-    if (handler._version != topic_manifest_version) {
+    if (handler._version != topic_manifest::first_version) {
         throw std::runtime_error(fmt_with_ctx(
           fmt::format,
           "topic manifest version {} is not supported",
           handler._version));
     }
+
+    _manifest_version = topic_manifest::first_version;
```

Comment:
just being explicit.
in the "wrong but not explicitly forbidden by c++" case of reusing a topic_manifest json,
decoding a serde version and then a json version, the second time around it would still report that it decoded a serde manifest